10-JavaScript/js-intro.md

```diff
@@ -0,0 +1,439 @@
+# Welcome to JavaScript!
+
+## Learning Goals
+
+- What is JavaScript? Why is it worth learning?
+- Discover the varying applications of JavaScript
+- Learn where Node and npm fit in all of this
+- Apply JavaScript naming conventions and comment syntax
+- Recognize the different data types in JavaScript
+- Understand that Objects are very different in JavaScript compared to Ruby
+
+## The Roots of JS
+
+JavaScript was originally developed in **ten days** in May 1995 by Brendan Eich, while he was working for Netscape Communications Corporation ([Wikipedia](https://en.wikipedia.org/wiki/JavaScript))
+
+Although it was developed under the name Mocha, the language was officially called LiveScript when it first shipped in beta releases of Netscape Navigator 2.0 in September 1995. It was renamed JavaScript when it was deployed in the Netscape browser version 2.0B3.
+
+### JavaScript != Java
+
+The change of name from LiveScript to JavaScript roughly coincided with Netscape adding support for _Java_ technology in its Netscape Navigator Web browser. The final choice of name caused confusion, giving the impression that the language was a spin-off of the Java programming language, and the choice has been characterized as a marketing ploy by Netscape to give JavaScript the cachet of what was then the hot new Web programming language.
+
+## Node is the name of the game
+
+[Node.js](https://nodejs.org/en/) is an open-source, cross-platform **runtime environment** for developing server-side Web applications. Node.js is not a JavaScript framework... It is a runtime environment that interprets JavaScript using Google's V8 JavaScript engine (same as what's in Chrome).
+
```

Comment:
Not sure how much of this you wrote vs was already here, but...

It would be good to make the parallel between the `node` program and the `ruby` program.